apps/prairielearn/src/components/EditQuestionPointsScore.html.ts

```diff
@@ -0,0 +1,131 @@
+import { escapeHtml, html } from '@prairielearn/html';
+
+import { AssessmentQuestion, InstanceQuestion } from '../lib/db-types.js';
+
+export function EditQuestionPointsScoreButton({
+  field,
+  instance_question,
+  assessment_question,
+  urlPrefix,
+  csrfToken,
+  buttonId,
+}: {
+  field: 'points' | 'auto_points' | 'manual_points' | 'score_perc';
+  instance_question: Omit<InstanceQuestion, 'modified_at'> & { modified_at: string };
+  assessment_question: AssessmentQuestion;
+  urlPrefix: string;
+  csrfToken: string;
+  buttonId: string;
+}) {
+  const editForm = EditQuestionPointsScoreForm({
+    field,
+    instance_question,
+    assessment_question,
+    urlPrefix,
+    csrfToken,
+    buttonId,
+  });
+  const label = {
+    points: 'points',
+    auto_points: 'auto points',
+    manual_points: 'manual points',
+    score_perc: 'score percentage',
+  }[field];
+
+  return html`<button
+    type="button"
+    class="btn btn-xs btn-secondary"
+    id="${buttonId}"
+    data-toggle="popover"
+    data-container="body"
+    data-html="true"
+    data-placement="auto"
+    title="Change question ${label}"
+    data-content="${escapeHtml(editForm)}"
+  >
+    <i class="fa fa-edit" aria-hidden="true"></i>
+  </button>`;
+}
+
+function EditQuestionPointsScoreForm({
+  field,
+  instance_question,
+  assessment_question,
+  urlPrefix,
+  csrfToken,
+  buttonId,
+}: {
+  field: 'points' | 'auto_points' | 'manual_points' | 'score_perc';
+  instance_question: Omit<InstanceQuestion, 'modified_at'> & { modified_at: string };
```

Comment:
Checking my understanding: this is because Postgres dates have microsecond precision but JavaScript dates only have millisecond precision? Presumably we could clean this up by moving this check from SQL into JS?

https://github.com/PrairieLearn/PrairieLearn/blob/c11b173b9cba8cdc25d3167d6b046922d78d19fb/apps/prairielearn/src/lib/manualGrading.sql#L382-L383

Reply:
Indeed. That would for sure be a valid alternative, but I'd prefer to do that in a separate PR.